### PR TITLE
[WIP][FEATURE] Observe changes through Extbase repositories

### PR DESCRIPTION
**Pull request by Mabahe**
Resolves: #126

### CODE REVIEW

**Comment by Mabahe** [comment]
If added the WIP patch as PR to make it more visible. Better description and docu is missing for the moment

**Comment by timohund** [comment]
@Mabahe Thanks for sharing and getting this started. I think it would be nice to get it into version 9.0 of EXT:solr. Do you have the time to add documentation? In addition it would be good to have this tested with our automated tests.

**Comment by dkd-kaehm** [comment]
Notice 4 myself : D16239

**Comment by WebsiteDeveloper** [comment]
@Mabahe @timohund any info on the progress of this PR would really love this feature

**Comment by WebsiteDeveloper** [comment]
@timohund I would love to contribute some Tests for this
Any pointers on where to start?

**Comment by timohund** [comment]
@WebsiteDeveloper Nice to here that. 
A good starting point is: 

https://github.com/TYPO3-Solr/ext-solr/blob/master/CONTRIBUTING.md#automated-tests
Or the .travis.yml in our project.

It is also possible to run the tests in a local travis docker container. If you need further assistance we can also talk on slack and maybe create a script for that.

**Comment by WebsiteDeveloper** [comment]
Alright i will check it out :)
Any thoughts on then backporting this feature to older versions?

**Comment by timohund** [comment]
As long at is not a breaking change i think in general we could backport that.

**Comment by dkd-kaehm** [comment]
@Mabahe 
Could you please refresh the things for dev-master(future 11) @ TYPO3 9.5 + 10.4?

**Comment by dkd-kaehm** [comment]
Rebased to current master. We should complete this.

File: Classes/IndexQueue/DomainObjectObserver.php [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+
+    /**
+     * Affects new and updated objects
+     *
+     * @param DomainObjectInterface $object
+     */
+    public function afterUpdateObject($object)
+    {
+        $forcedChangeTime = 0;
+        list($_, $object, $forcedChangeTime) = $this->signalSlotDispatcher->dispatch(__CLASS__, 'preUpdateCheck', [$this, $object, $forcedChangeTime]);
```
Comment (dmitryd) [inline Classes/IndexQueue/DomainObjectObserver.php:139]: This should be:

```php
[ , $object, $forcedChangeTime] = ...
```

**Comment by dmitryd** [comment]
What about handling `afterInsertObject` for new records?

**Comment by Mabahe** [comment]
I'm on updating this.

**Comment by Mabahe** [comment]
@dmitryd @varioous new records are also affected by the updated event: `backend::persistObjects` calls `persistObject` calls `updateObject`. This is because `insertObject` inserts only the row without the relations; relations are handled in `updateObject` later on. Also the `Queue::updateItem` does an insert, if the item is missing. So IMO no extra handling for new records is needed. If I remember right it worked for new objects, or do you have noticed problems with new objects?

**Comment by dmitryd** [comment]
@Mabahe Cannot remember, unfortunately. I have no code to test this right now. :(

**Comment by Mabahe** [comment]
I will drop my branch https://github.com/Mabahe/ext-solr/tree/feature/extbase-domain-observer because of the bad name (branch missing). Also this patch is not compatible with the base branch. 

So will close this PR in favour of a new one using the right branch name.

**Pull request closed without merging**